src/Glpi/Controller/ItemType/Form/ContactFormController.php

```diff
@@ -33,42 +32,41 @@
  * ---------------------------------------------------------------------
  */
 
-if (!isset($_GET["id"])) {
-    $_GET["id"] = "";
-}
+namespace Glpi\Controller\ItemType\Form;
 
-if (!isset($_GET["withtemplate"])) {
-    $_GET["withtemplate"] = "";
-}
+use Contact;
+use Glpi\Controller\GenericFormController;
+use Glpi\Routing\Attribute\ItemtypeFormRoute;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+class ContactFormController extends GenericFormController
+{
+    #[ItemtypeFormRoute(Contact::class)]
+    public function __invoke(Request $request): Response
+    {
+        $request->attributes->set('class', Contact::class);
+
+        if ($request->query->has('getvcard')) {
+            return $this->generateVCard($request);
+        }
 
-$savedsearch = new SavedSearch();
-if (isset($_POST["add"])) {
-   //Add a new saved search
-    $savedsearch->check(-1, CREATE, $_POST);
-    if ($savedsearch->add($_POST)) {
-        if ($_SESSION['glpibackcreated']) {
-            Html::redirect($savedsearch->getLinkURL());
+        return parent::__invoke($request);
+    }
+
+    private function generateVCard(Request $request): Response
+    {
+        $id = $request->query->getInt('id', 1);
+
+        if ($id < 0) {
+            return new RedirectResponse($request->getBasePath() . '/front/contact.php');
```

Comment:
Not sure why it was just a redirection, but if the id is not passed to the request, then it is a bad request.

```suggestion
        $id = $request->query->getInt('id');

        if ($id === 0) {
            return new \Glpi\Exception\Http\BadRequestHttpException();
```